### PR TITLE
Add filename and lineNumber to scope

### PR DESCRIPTION
**Pull request by imaustink**
Related to #304

### CODE REVIEW

File: can-stache.js [hunk 1]
```diff
@@ -409,13 +406,13 @@ function stache (filename, template) {
 				makeRendererAndUpdateSection(state.textContentOnly || section, mode, expression, lineNo);
 			}
 		},
-		comment: function( text ) {
+		comment: function(text, lineNo) {
```
Comment (phillipskevin) [inline can-stache.js]: is this needed?

File: can-stache.js [hunk 2]
```diff
@@ -409,13 +406,13 @@ function stache (filename, template) {
...
 			// create comment node
 			section.add({
 				comment: text
 			});
 		},
-		done: function(){}
+		done: function(lineNo){}
```
Comment (phillipskevin) [inline can-stache.js]: is this needed?

File: test/stache-test.js [hunk 3]
```diff
@@ -6580,6 +6580,21 @@ function makeTest(name, doc, mutation) {
 		equal(warn1(), 0);
 		equal(warn2(), 0);
 	});
+	
+	test("scope has lineNumber", function(){
```
Comment (phillipskevin) [inline test/stache-test.js]: this should be a dev only test.

File: test/stache-test.js [hunk 4]
```diff
@@ -6580,6 +6580,21 @@ function makeTest(name, doc, mutation) {
...
+		var template = stache('<p>{{scope.lineNumber}}</p>\n<p>{{scope.lineNumber}}</p>');
```
Comment (phillipskevin) [inline test/stache-test.js]: this should be a dev only test.

File: src/mustache_core.js [hunk 5]
```diff
@@ -406,7 +413,7 @@ var core = {
 	 * @param {can.stache.Expression} expression
 	 * @param {Object} state The state of HTML where the expression was found.
 	 */
-	splitModeFromExpression: function(expression, state){
+	splitModeFromExpression: function(expression, state, lineNo){
```
Comment (phillipskevin) [inline src/mustache_core.js]: is this change needed?

File: can-stache.js [hunk 6]
```diff
@@ -424,7 +421,11 @@ function stache (filename, template) {
...
 			assign( optionsScope.inlinePartials, inlinePartials );
 		}
+
 		scope.set('scope.view', scopifiedRenderer);
+		//!steal-remove-start
+		scope.set('scope.filename', section.filename);
+		//!steal-remove-end
 
 		// pass filename so it can be used in dev warnings
 		optionsScope._meta.filename = section.filename;
```
Comment (phillipskevin) [inline can-stache.js]: do we still need the filename added to `optionsScope._meta`?

Comment (imaustink) [reply can-stache.js]: I believe it is not.
